pkg/shell/terraform.go

```diff
@@ -135,17 +137,63 @@ func outputModule(tf *tfexec.Terraform) (map[string]cty.Value, error) {
 	return outputValues, nil
 }
 
-// note planned deprecration of Plan in favor of JSON-only format
-// may need to determine future-proof way of getting human-readable plan
-// https://github.com/hashicorp/terraform-exec/blob/1b7714111a94813e92936051fb3014fec81218d5/tfexec/plan.go#L128-L129
+// See https://github.com/hashicorp/terraform/blob/4ce385a19b93cf7f1b7780d9b2d3cadc5d0ddb31/internal/command/views/json/diagnostic.go#L34
+type Diagnostic struct {
+	Severity string `json:"severity"`
+	Summary  string `json:"summary"`
+	Detail   string `json:"detail"`
+}
+
+type JsonMessage struct {
+	Level      string     `json:"@level"`
+	Diagnostic Diagnostic `json:"diagnostic"`
+}
+
+func parseJsonMessages(data string) []JsonMessage {
+	res := []JsonMessage{}
+	// Every message is on a single line
+	sc := bufio.NewScanner(strings.NewReader(data))
+	for sc.Scan() {
+		var msg JsonMessage
+		if err := json.Unmarshal([]byte(sc.Text()), &msg); err != nil {
+			continue // silently skip the message
+		}
+		res = append(res, msg)
+	}
+	return res
+}
+
+func helpOnPlanError(msgs []JsonMessage) string {
```

Comment:
Add a comment here that the function is enforcing the logic that the only undefined variables can possibly come from IGC references. This may not always be true.